### PR TITLE
Change response to config/status REST interface

### PR DESCRIPTION
**Pull request by cdjackson**
I don't think a "404 Not Found" response is appropriate when a binding doesn't support the config status interface. Instead, the interface should respond with an empty config status.
In chrome at least, a 404 is treated as an error, and is logged in such a way. In my opinion, this is not an error - it simply means that the binding isn't supporting this interface and since there's no way to find this out (right?) we shouldn't be producing errors.

This change returns 404 if the thing doesn't exist. If the config status doesn't exist, it returns an empty set.

Signed-off-by: Chris Jackson <chris@cd-jackson.com>

### CODE REVIEW

File: bundles/io/org.eclipse.smarthome.io.rest.core/src/main/java/org/eclipse/smarthome/io/rest/core/thing/ThingResource.java [hunk 1]
```diff
@@ -423,14 +424,24 @@ public Response updateConfiguration(@HeaderParam(HttpHeaders.ACCEPT_LANGUAGE) St
...
-            @ApiResponse(code = 404, message = "Config status for thing not found.") })
+            @ApiResponse(code = 404, message = "Thing not found.") })
     public Response getConfigStatus(@HeaderParam(HttpHeaders.ACCEPT_LANGUAGE) String language,
             @PathParam("thingUID") @ApiParam(value = "thing") String thingUID) throws IOException {
+        ThingUID thingUIDObject = new ThingUID(thingUID);
+
+        // ask whether the Thing exists at all, 404 otherwise
+        Thing thing = thingRegistry.get(thingUIDObject);
+        if (null == thing) {
+            logger.info("Received HTTP PUT request for thing config status at '{}' for the unknown thing '{}'.",
```
Comment (maggu2810) [inline bundles/io/org.eclipse.smarthome.io.rest.core/src/main/java/org/eclipse/smarthome/io/rest/core/thing/ThingResource.java]: "HTTP PUT request" should be a "HTTP GET request".

**Comment by maggu2810** [comment]
Ok for me

File: bundles/io/org.eclipse.smarthome.io.rest.core/src/main/java/org/eclipse/smarthome/io/rest/core/thing/ThingResource.java (hunk 1 above)
Comment (cdjackson) [inline bundles/io/org.eclipse.smarthome.io.rest.core/src/main/java/org/eclipse/smarthome/io/rest/core/thing/ThingResource.java]: Updated...

**Comment by cdjackson** [comment]
Updated to remove conflict after the refactoring

**Pull request merged**